manga_translator/rendering/text_render.py

```
@@ -120,8 +120,23 @@ def normalize_vertical_ellipsis_text(text: str) -> str:
     return text or ''
 
 
-def auto_add_horizontal_tags(text: str) -> str:
-    if not text or '<H>' in text or '<h>' in text.lower():
+def auto_add_horizontal_tags_symbols(text: str) -> str:
+    """只处理标点符号合并：!! / !? / ！？ 等连续标点包裹为 <H> 标签。"""
+    if not text:
+        return text
+    seg = re.sub(r'[!?！？]{2,4}', r'<H>\g<0></H>', text)
+    pair_re = re.compile(r'<H>([!?！？]{2,4})</H>\s*(\r\n|\r|\n|\[BR\]|<br\s*/?>|【BR】)\s*<H>([!?！？]{2,4})</H>', re.IGNORECASE)
```

Ruff (0.15.15)

[warning] String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)